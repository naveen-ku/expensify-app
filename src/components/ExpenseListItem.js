import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  return (
    <div>
      <h2>
        <Link to={`edit/${id}`}>{description}</Link>
      </h2>
      <p>
        {amount}-{createdAt}
      </p>
    </div>
  );
};

export default ExpenseListItem;
