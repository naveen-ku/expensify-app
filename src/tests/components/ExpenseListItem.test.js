import React from "react";
import ExpenseListItem from "../../components/ExpenseListItem";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import toJSON from "enzyme-to-json";

test("should render expense list items with details", () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
