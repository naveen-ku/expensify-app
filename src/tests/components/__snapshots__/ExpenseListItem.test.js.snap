// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list items with details 1`] = `
<div>
  <h2>
    <[object Object]
      to="edit/1"
    >
      Water
    </[object Object]>
  </h2>
  <p>
    300
    -
    0
  </p>
</div>
`;
